### DISCUSSION TITLE
Help with Sending Emails from Supabase when New Row is Added

### DISCUSSION THREAD

**Discussion by Adamuser14** [supabase/supabase#30764] [Questions]
Hey everyone,

I'm relatively new to coding and databases, but I'm working on a small project for school, and I need to get it done as quickly as possible. I'm using Supabase as the backend for my FlutterFlow project, and I need to automate the process of sending an email when a new row is added to a specific table.

In my table, I have a column called `email`, and I want the system to send an email to this address when a new row is added. The email should include dynamic content, such as the `otp` column value and other fields from the row, and I need the email subject and body to reflect this.

I’ve tried several no-code automation tools, but most are either too expensive or have a long delay (15 minutes) between the event and the email being sent. I can’t afford to make my users wait that long, so I’m looking for a solution that can send the email immediately after a new row is inserted.

Could anyone help with this in a simple, step-by-step way? I'd greatly appreciate any guidance, as I'm still learning and need to get this project up and running soon.

Thanks so much! 

![Screenshot_10](https://github.com/user-attachments/assets/a3500302-23f9-4b03-8795-9e968a556198)

**Comment by GaryAustin1** [comment]
You would use a trigger function or webhook to make a call to your (a) SMTP provider API.  You can use an edge function to send the email or even use the http or pg_net extension to do it from the database if there is an api.   I would recommend the edge function route.  https://supabase.com/docs/guides/functions/examples/send-emails

**Reply by Adamuser14** [reply]
Thank you for your response!
I’ve gone through the documentation, but I’m still unclear about where to add the code and how to configure specific fields to be sent with the email from a specefic table . Additionally, I’m not very familiar with how HTTP requests work or how to use Supabase extensions as i tried really hard to find something with no code or the lowest code possible also im using supabase on the web im not hosting it locally or in my server, I apologize for the confusion and appreciate your guidance!

**Reply by GaryAustin1** [reply]
You are more likely to get help if you ask specific questions when you run into an issue.
https://supabase.com/docs/guides/database/webhooks
https://www.youtube.com/watch?v=dJk2wbdLDpE

I just searched the web to get those I saw lots of other hits too that might be useful as guides.

**Reply by Adamuser14** [reply]
Thank you so much, ill check them out and i'll try to learn how these works, i had hard time trying to know how webhooks, api stuff works
